content/en/blog/_posts/2024-xx-xx-csi-changed-block-tracking.md

```diff
@@ -0,0 +1,128 @@
+- _High resource utilization_: Backup operations consume substantial network bandwidth and I/O resources, especially for large data volumes and data-intensive applications.
+- _Increased storage costs_: Repetitive full backups store redundant data, causing storage requirements to grow linearly even when only a small percentage of data actually changes between backups.
+
+The Changed Block Tracking API addresses these challenges by providing native Kubernetes support for incremental backup capabilities through the CSI interface.
+
+## Key components
+
+The implementation consists of three primary components:
+
+1. _CSI SnapshotMetadata Service API_: A gRPC service that provides information about volume snapshots, including changed block data
+2. _SnapshotMetadataService Custom Resource_: Advertises the availability of a CSI driver's metadata service along with connection details
+3. _External Snapshot Metadata Sidecar_: Implements the server-side of the API, bridging CSI drivers with backup applications
+
+## Implementation requirements
+
+### Storage provider responsibilities
+
+Storage vendors who want to support the changed block tracking feature must implement specific requirements:
+
+1. _Implement CSI RPCs_: Storage vendors need to implement the `SnapshotMetadata` service as defined in the [CSI specifications protobuf](https://github.com/container-storage-interface/spec/blob/master/csi.proto). This service requires server-side streaming implementations for the following RPCs:
+
+   - `GetMetadataAllocated`: For identifying allocated blocks in a snapshot
+   - `GetMetadataDelta`: For determining changed blocks between two snapshots
+
+2. _Storage backend capabilities_: Ensure the storage backend has the capability to track and report block-level changes.
+
+3. _Deploy external components_: Integrate with the `external-snapshot-metadata` sidecar to expose the snapshot metadata service.
+
+4. _Register custom resource_: Create and maintain a `SnapshotMetadataService` custom resource that advertises the availability of their metadata service and provides connection details.
+
+5. _Support error handling_: Implement proper error handling for these RPCs according to the CSI specification requirements.
+
+### Backup client responsibilities
+
+Backup vendors looking to leverage this feature must:
+
+1. **Set up authentication**: Configure proper authentication for clients accessing the storage system using Kubernetes ServiceAccount tokens.
+
+2. **Implement streaming client-side code**: Develop clients that implement the streaming gRPC APIs defined in the [schema.proto](https://github.com/kubernetes-csi/external-snapshot-metadata/blob/main/proto/schema.proto) file. Specifically:
+   - Implement streaming client code for `GetMetadataAllocated` and `GetMetadataDelta` methods
+   - Handle server-side streaming responses efficiently as the metadata comes in chunks
+   - Process the `SnapshotMetadataResponse` message format with proper error handling
```

Comment:
I suggest adding that the external-snapshot-metadata repository provides a convenient [iterator](https://github.com/kubernetes-csi/external-snapshot-metadata/tree/master/pkg/iterator) support package that handles all of the above.